GridTariffAPI.lib/Services/Helpers/ServiceHelper.cs

```diff
@@ -12,6 +15,82 @@ public ServiceHelper(GridTariffApiConfig gridTariffApiConfig)
 
         }
 
+        public List<TimePeriod> GetMonthPeriods(DateTimeOffset fromDate, DateTimeOffset toDate, IReadOnlyList<int> months)
+        {
+            var accumulator = new List<TimePeriod>();
+            var fromDateLocaled = ToConfiguredTimeZone(fromDate);
+            var toDateLocaled = ToConfiguredTimeZone(toDate);
+
+            //accumulate
+            while (fromDateLocaled < toDateLocaled)
+            {
+                if (months.Contains(fromDateLocaled.Month))
+                {
+                    var monthEndLocaled = fromDateLocaled.AddDays(1 - fromDateLocaled.Day).AddMonths(1);
+                    monthEndLocaled = monthEndLocaled.AddHours(-monthEndLocaled.Hour).AddMinutes(-monthEndLocaled.Minute);
+                    var timePeriod = new TimePeriod();
+                    timePeriod.StartDate = fromDateLocaled;
+                    timePeriod.EndDate = monthEndLocaled < toDateLocaled ? monthEndLocaled : toDateLocaled;
+
+                    accumulator.Add(timePeriod);
+                }
+                fromDateLocaled = fromDateLocaled.AddDays(1 - fromDateLocaled.Day).AddMonths(1);
+                fromDateLocaled = fromDateLocaled.AddHours(-fromDateLocaled.Hour).AddMinutes(-fromDate.Minute);
+            }
+
+            //adjust for standard time/DST
+            foreach (var element in accumulator)
+            {
+                element.StartDate = CreateLocaledDateTimeOffset(element.StartDate);
+                element.EndDate = CreateLocaledDateTimeOffset(element.EndDate);
+            }
+
+            //concat
+            var retVal = new List<TimePeriod>();
+            accumulator = accumulator.OrderBy(x => x.StartDate).ToList();
+            if (accumulator.Count > 0)
+            {
+                var accTimePeriod = accumulator[0];
+                for (int i =1;i< accumulator.Count;i++)
+                {
+                    var currTimePeriod = accumulator[i];
+                    if (accTimePeriod.EndDate.Year == currTimePeriod.StartDate.Year && accTimePeriod.EndDate.Month == currTimePeriod.StartDate.Month)
+                    {
+                        accTimePeriod.EndDate = currTimePeriod.EndDate;
+                    }
+                    else
+                    {
+                        retVal.Add(accTimePeriod);
+                        accTimePeriod = currTimePeriod;
+                    }
+                }
+                retVal.Add(accTimePeriod);
+            }
+
+            return retVal;
+        }
+
+        public DateTimeOffset CreateLocaledDateTimeOffset(DateTimeOffset value)
```

Comment:
At denne tar inn DateTimeOffset, men ikke bryr seg om offset'en virker merkelig. Men kanskje ikke behov for denne metoden og den under hvis man legger om beregningen av månedsovergang fra å regne om gitt tidspunkt til å angi dag 1.

Reply:
Formålet med denne funksjonen er å ta dato/tid i konfigurert tidssone og sette korrekt offset for å hensynta at offset kan endre seg når man legger til en måned.
Vi kan nok fjerne denne om vi klarer å håndere offset korrekt i den nye funksjonen som opprettes med grunnlag i første kommentar.

Reply:
Funksjon renamet og lagt inn kommentar om hva denne gjør.